library/std/src/net/ip/tests.rs

```diff
@@ -593,6 +600,12 @@ fn ipv6_properties() {
 
     check!("1::", &[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], global | unicast_global);
 
+    check!(
+        "::ffff:127.0.0.1",
+        &[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xff, 0xff, 0x7f, 0, 0, 1],
+        global | unicast_global | ipv4_mapped
```

Comment:
According to the [IANA IPv6 Special Address Registry](https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml) IPv4-mapped addresses are not globally reachable. However due to the current simplistic implementation of `Ipv6Addr::is_global` compared to `Ipv4Addr::is_global`, Rust currently considers these addresses global. I plan to address this in a future PR reworking `is_global`.

Reply:
The rework of `Ipv6Addr::is_global` has been submitted (#86634).